akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -249,14 +269,35 @@ private[cluster] case class HeartbeatNodeRing(selfAddress: UniqueAddress, nodes:
   /**
    * The receivers to use from a specified sender.
    */
-  def receivers(sender: UniqueAddress): immutable.Set[UniqueAddress] =
+  def receivers(sender: UniqueAddress): Set[UniqueAddress] =
     if (useAllAsReceivers)
       nodeRing - sender
     else {
-      val slice = nodeRing.from(sender).tail.take(monitoredByNrOfMembers)
-      if (slice.size < monitoredByNrOfMembers)
-        (slice ++ nodeRing.take(monitoredByNrOfMembers - slice.size))
-      else slice
+
+      // Pick nodes from the iterator until n nodes that are not unreachable have been selected.
+      // Intermediate unreachable nodes are also included in the result.
```

Comment:
> why do we include them?

It might actually be possible to just skip the unreachable here. The once that we have been monitoring before will be continued to be monitored anyway until they are reachable or removed. That is taken care of by another mechanism (`oldReceiversNowUnreachable`).